app/forms/NirmsNumberFormProvider.scala

```diff
@@ -27,7 +26,24 @@ class NirmsNumberFormProvider @Inject() extends Mappings {
   def apply(): Form[String] =
     Form(
       "value" -> text("nirmsNumber.error.required")
-        .transform(removeWhitespace, identity[String])
+        .transform(removeSpaces, identity[String])
         .verifying(regexp(StringFieldRegex.nirmsRegex, "nirmsNumber.error.invalidFormat"))
+        .transform(addHyphens, identity[String])
     )
+
+  def removeSpaces: String => String = _.replaceAll(" ", "")
+  def addHyphens: String => String = { original =>
+    val length = 11
+    if (original.length == length) {
```

Comment:
just a tip, you may want to look into patch here: e.g. 

"abdet67867".patch(3, "-", 0).patch(5, "-", 0)
> val res3: String = abd-e-t67867